### PR TITLE
Improve errors

### PR DESCRIPTION
**Pull request by crackboy88**
**Summary**
I have improved some error messages in minimize.cpp and neighbor.cpp.

**Related Issue(s)**

**Author(s)**
Jiancheng Chen

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

**Implementation Notes**

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [ ] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

**Further Information, Files, and Links**

### CODE REVIEW

File: src/neighbor.cpp [hunk 1]
```diff
@@ -2582,60 +2580,58 @@ void Neighbor::modify_params(int narg, char **arg)
   int iarg = 0;
   while (iarg < narg) {
     if (strcmp(arg[iarg],"every") == 0) {
-      if (iarg+2 > narg) error->all(FLERR,"Illegal neigh_modify command");
+      if (iarg+2 > narg) error->all(FLERR, "Invalid every keyword: expected 1 every argument");
```
Comment (akohlmey) [inline src/neighbor.cpp]: Unlike what we discussed with the `narg !=` test case. This is a case where you *should* be using `utils::missing_cmd_args()`.

File: src/neighbor.cpp [hunk 2]
```diff
@@ -2582,60 +2580,58 @@ void Neighbor::modify_params(int narg, char **arg)
...
       every = utils::inumeric(FLERR,arg[iarg+1],false,lmp);
-      if (every <= 0) error->all(FLERR,"Illegal neigh_modify command");
+      if (every <= 0) error->all(FLERR, "Invalid every argument: {}", every);
```
Comment (akohlmey) [inline src/neighbor.cpp]: This should be "Invalid neigh_modify every argument: {}"
Same for multiple cases below

File: src/neighbor.cpp [hunk 3]
```diff
@@ -2582,60 +2580,58 @@ void Neighbor::modify_params(int narg, char **arg)
...
-      if (iarg+2 > narg) error->all(FLERR,"Illegal neigh_modify command");
+      if (iarg+2 > narg) error->all(FLERR, "Invalid include keyword: expected 1 include argument");
       includegroup = group->find(arg[iarg+1]);
-      if (includegroup < 0)
-        error->all(FLERR,"Invalid group ID in neigh_modify command");
-      if (includegroup && (atom->firstgroupname == nullptr ||
-                            strcmp(arg[iarg+1],atom->firstgroupname) != 0))
-        error->all(FLERR,
-                   "Neigh_modify include group != atom_modify first group");
+      if (includegroup == -1)
```
Comment (akohlmey) [inline src/neighbor.cpp]: Please don't change the logic. `includegroup < 0` is more defensive programming (and won't make a difference in practice).

File: src/neighbor.cpp [hunk 4]
```diff
@@ -2582,60 +2580,58 @@ void Neighbor::modify_params(int narg, char **arg)
...
+        error->all(FLERR, "Invalid include keyword: group {} not found", arg[iarg+1]);
+      if (atom->firstgroupname == nullptr)
+          error->all(FLERR, "Invalid include keyword: atom_modify first command must be used");
+      if (strcmp(arg[iarg+1],atom->firstgroupname) != 0)
+        error->all(FLERR, "Neigh_modify include group != atom_modify first group: {}", atom->firstgroupname);
       iarg += 2;
-
     } else if (strcmp(arg[iarg],"exclude") == 0) {
-      if (iarg+2 > narg) error->all(FLERR,"Illegal neigh_modify command");
+      if (iarg+2 > narg) error->all(FLERR, "Invalid exclude keyword: loss exclude arguments");
```
Comment (akohlmey) [inline src/neighbor.cpp]: I don't know what this message means, but see my previous comment about using `utils::missing_cmd_args()` for all these cases where arguments are missing.

File: src/neighbor.cpp [hunk 5]
```diff
@@ -2676,30 +2672,28 @@ void Neighbor::modify_params(int narg, char **arg)
...
           ex_mol_intra[nex_mol] = 0;
         nex_mol++;
         iarg += 3;
-
       } else if (strcmp(arg[iarg+1],"none") == 0) {
         nex_type = nex_group = nex_mol = 0;
         iarg += 2;
-
-      } else error->all(FLERR,"Illegal neigh_modify command");
+      } else error->all(FLERR,"Invalid exclude keyword: {} argument doesn't exist", arg[iarg+1]);
```
Comment (akohlmey) [inline src/neighbor.cpp]: Better to write here: "Unknown neigh_modify exclude keyword: {}"

File: src/neighbor.cpp [hunk 6]
```diff
@@ -2774,7 +2768,7 @@ void Neighbor::modify_params(int narg, char **arg)
       }
 
       iarg += 2 + ncollections;
-    } else error->all(FLERR,"Illegal neigh_modify command");
+    } else error->all(FLERR,"Illegal neigh_modify command: {} keyword doesn't exist", arg[iarg]);
```
Comment (akohlmey) [inline src/neighbor.cpp]: See similar comment about: This would better be "Unknown neigh_modify keyword: {}"

File: src/neighbor.cpp [hunk 7]
```diff
@@ -2956,4 +2948,4 @@ double Neighbor::memory_usage()
   if (neigh_improper) bytes += neigh_improper->memory_usage();
 
   return bytes;
-}
+}
```
Comment (akohlmey) [inline src/neighbor.cpp]: Please make certain that there is a newline character at the end of the file (check your editor settings, if necessary).

**Review by akohlmey — CHANGES_REQUESTED** [review]
@crackboy88 thank your very much for your submission and for being the first Code Clinic participant to do so. :+1: 

I've added some comments. Specifically to individual lines, but keep in mind that in some cases the same issue is repeated multiple times and thus should be addressed multiple times.

**Review by akohlmey — APPROVED** [review]

**Review by athomps — APPROVED** [review]
Looks great!

**Pull request merged**